src/Orleans/Streams/PersistentStreams/IStreamQueueBalancer.cs

```diff
@@ -53,9 +53,8 @@ public interface IStreamQueueBalancer
     /// indicating that the balance of queues has changed.
     /// It should be implemented by components interested in stream queue load balancing.
     /// When change notification is received, listener should request updated list of queues from the queue balancer.
-    /// This interface inherit from IAddressable for threading-safe concern
     /// </summary>
-    public interface IStreamQueueBalanceListener :IAddressable
+    public interface IStreamQueueBalanceListener
```

Comment:
> "This interface inherit from IAddressable for threading-safe concern"

It inherits from IAddressable so it can be implemented by system target and notification be called as grain call. The right and proper way to talk between components inside Orleans is via grain (system target) calls. Except for some small number of limited cases when we break isolation. But here is not such a case and thus I don't see a reason to avoid using grain calls for cross component communication and insist on breaking isolation (I call "invoking function directly, even if after that we enqueue an Task directly" as breaking isolation).